kibana-notebooks/server/adaptors/default_backend.ts

```diff
@@ -304,32 +308,53 @@ export class DefaultBackend implements NotebookAdaptor {
    * Currently only runs markdown by copying input.inputText to result
    * UI renders Markdown
    */
-  runParagraph = async function (paragraphs: Array<DefaultParagraph>, paragraphId: string) {
+  runParagraph = async function (paragraphs: Array<DefaultParagraph>, paragraphId: string, client: ILegacyScopedClusterClient) {
     try {
       const updatedParagraphs = [];
-      paragraphs.map((paragraph: DefaultParagraph) => {
-        const updatedParagraph = { ...paragraph };
-        if (paragraph.input.inputType === 'MARKDOWN' && paragraph.id === paragraphId) {
-          updatedParagraph.dateModified = new Date().toISOString();
-          updatedParagraph.output = [
-            {
-              outputType: 'MARKDOWN',
-              result: paragraph.input.inputText,
-              execution_time: '0s',
-            },
-          ];
-        } else if (paragraph.input.inputType === 'VISUALIZATION' && paragraph.id === paragraphId) {
+      let index = 0;
+      for (index = 0; index < paragraphs.length; ++index) {
+        const updatedParagraph = { ...paragraphs[index] };
+        if (paragraphs[index].input.inputType === 'MARKDOWN' && paragraphs[index].id === paragraphId) {
           updatedParagraph.dateModified = new Date().toISOString();
-          updatedParagraph.output = [
-            {
-              outputType: 'VISUALIZATION',
-              result: '',
-              execution_time: '0s',
-            },
-          ];
+          if (inputIsQuery(paragraphs[index].input.inputText)) {
+            const queryService = new QueryService(client);
+            let output = await getQueryOutput(paragraphs[index].input.inputText, queryService);
+            updatedParagraph.output = [
+              {
+                outputType: 'QUERY',
+                result: output.data.resp,
+                execution_time: '0s',
```

Comment:
why does the execution time all 0s? Can we capture the actual or approximate run time instead?

Reply:
I will create an issue for it, but looks like the `execution_time` has always been hard-coded to '0s', I don't think that change is in the scope of this PR